services/horizon/internal/integration/invokehostfunction_test.go

```diff
@@ -24,13 +26,42 @@ const increment_contract = "soroban_increment_contract.wasm"
 // Refer to ./services/horizon/internal/integration/contracts/README.md on how to recompile
 // contract code if needed to new wasm.
 
-func TestContractInvokeHostFunctionInstallContract(t *testing.T) {
+func TestInvokeHostFns(t *testing.T) {
+	// first test contracts when soroban processing is enabled
+	DisabledSoroban = false
+	runAllTests(t)
+	// now test same contracts when soroban processing is disabled
+	DisabledSoroban = true
+	runAllTests(t)
+}
+
+func runAllTests(t *testing.T) {
+	tests := []struct {
+		name string
+		fn   func(*testing.T)
+	}{
+		{"CaseContractInvokeHostFunctionInstallContract", CaseContractInvokeHostFunctionInstallContract},
+		{"CaseContractInvokeHostFunctionCreateContractByAddress", CaseContractInvokeHostFunctionCreateContractByAddress},
+		{"CaseContractInvokeHostFunctionInvokeStatelessContractFn", CaseContractInvokeHostFunctionInvokeStatelessContractFn},
+		{"CaseContractInvokeHostFunctionInvokeStatefulContractFn", CaseContractInvokeHostFunctionInvokeStatefulContractFn},
+	}
+
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf("Soroban Processing Disabled = %v. ", DisabledSoroban)+tt.name, func(t *testing.T) {
+			tt.fn(t)
+		})
+	}
+}
+
+func CaseContractInvokeHostFunctionInstallContract(t *testing.T) {
 	if integration.GetCoreMaxSupportedProtocol() < 20 {
 		t.Skip("This test run does not support less than Protocol 20")
 	}
 
 	itest := integration.NewTest(t, integration.Config{
-		ProtocolVersion:  20,
+		ProtocolVersion: 20,
+		HorizonEnvironment: map[string]string{
+			"DISABLE_SOROBAN_INGEST_PROCESSORS": fmt.Sprint(DisabledSoroban)},
```

Comment:
does sprint work here w/o the formatting specifier? I thought you needed
```suggestion
			"DISABLE_SOROBAN_INGEST_PROCESSORS": strconv.FormatBool(DisabledSoroban)},
```

Reply:
yes, it outputs stringified true/false

Reply:
neat, thanks !